gui/next/src/lib/api/record.js

```diff
@@ -184,6 +136,94 @@ const buildPropertiesFilter = (filters = []) => {
 
 
 
+// purpose:		builds all the needed data to build and trigger GraphQL mutation
+// arguments: FormData object with the column, type and value (column[value], column[type])
+// returns:   variablesDefinition (string) - GraphQL variables definitions $variable: Type
+//            variables (object) - values for defined variables passed to GraphQL API
+//            properties (string) - list of GraphQL properties for given variables to use inside "properties: []"
+// ------------------------------------------------------------------------
+const buildMutationIngredients = (formData) => {
+
+  // entries from the form (object)
+  const formEntries = formData.entries();
+  // graphql variables definition for the query (string)
+  let variablesDefinition = '';
+  // variables passed with the query (object)
+  let variables = {};
+  // properties list with variables as their value passed with the query (string)
+  let properties = '';
+  // helper object that will store the column name with all it's corresponding propertiest needed to pass to graphql (object)
+  let columns = {};
+
+  // values we are getting from FormData are strings, this will parse them depending on the column type we are taking them from
+  function parseValue(type, value){
```

Comment:
@z-x I don't get why you have defined function inside function? It can be defined on the same level as `buildMutationIngredients`

Reply:
Cause this parsing is relevant only to that outer function, I see no disadvantages and it's an easy way to mark relations between those two, are there any?